server/src/utils/mixerConnections/AtemConnection.ts

```diff
@@ -0,0 +1,415 @@
+import { Atem, AtemState } from 'atem-connection'
+
+import { store, state } from '../../reducers/store'
+import {
+    fxParamsList,
+    IMixerProtocol,
+} from '../../../../shared/src/constants/MixerProtocolInterface'
+import { logger } from '../logger'
+import { storeSetMixerOnline } from '../../../../shared/src/actions/settingsActions'
+import { IFader } from '../../../../shared/src/reducers/fadersReducer'
+import { IChannel } from '../../../../shared/src/reducers/channelsReducer'
+import { dbToFloat, floatToDB } from './LawoRubyConnection'
+import {
+    AudioMixOption,
+    FairlightAudioMixOption,
+    FairlightInputConfiguration,
+} from 'atem-connection/dist/enums'
+import {
+    storeFaderLevel,
+    storeInputGain,
+    storeSetMute,
+    storeSetPgm,
+    storeSetVo,
+} from '../../../../shared/src/actions/faderActions'
+import { storeSetChLabel } from '../../../../shared/src/actions/channelActions'
+import { FairlightAudioSource } from 'atem-connection/dist/state/fairlight'
+
+export class AtemMixerConnection {
+    private _connection: Atem
+
+    private _chNoToSource: Record<number, number> = {}
+    private _sourceToChNo: Record<number, number> = {}
+    private _sourceTracks: Record<number, '-65280' | '-255' | '-256'> = {}
+    private _firstConnection = true
+
+    constructor(
+        private mixerProtocol: IMixerProtocol,
+        public mixerIndex: number
+    ) {
+        this._connection = new Atem()
+
+        this._connection.on('connected', () => {
+            logger.info('Atem connected')
+            store.dispatch(storeSetMixerOnline(this.mixerIndex, true))
+            global.mainThreadHandler.updateMixerOnline(this.mixerIndex)
+
+            this.setupMixerConnection()
+        })
+        this._connection.on('disconnected', () => {
+            logger.info('Atem disconnected')
+            store.dispatch(storeSetMixerOnline(this.mixerIndex, false))
+            global.mainThreadHandler.updateMixerOnline(this.mixerIndex)
+        })
+
+        this._connection
+            .connect(state.settings[0].mixers[this.mixerIndex].deviceIp)
+            .catch(() => {
+                logger.error(
+                    'Failed to connect to atem ' +
+                        state.settings[0].mixers[this.mixerIndex].deviceIp
+                )
+            })
```

Comment:
Just to give some more information, the caught error could be attached to the log:
```suggestion
            .catch((error) => {
                logger.data(error).error(
                    'Failed to connect to atem ' +
                        state.settings[0].mixers[this.mixerIndex].deviceIp
                )
            })
```